src/ds_memory_llrbt.c

```diff
@@ -74,6 +74,10 @@ struct memory_node_s* ds_memory_llrbt_create_node(address_3bc_t address)
     struct memory_node_s* new_node
         = (struct memory_node_s*)malloc(sizeof(struct memory_node_s));
 
+    if (new_node == NULL) {
+        return NULL;
```

Comment:
please use error handler
```c
if (new_node == NULL) {
   driver_program_error(app, ERROR_OUT_OF_MEMORY);
}
```

Reply:
Sorry, but I think `app` is not defined here.

Reply:
> Sorry, but I think `app` is not defined here.

okay, it would have to be a more thought out solution.

but I recommend not working on it, this module will be completely redone or deleted...

it is as a legacy in the branch [`develop-wip-wip`](https://github.com/RodrigoDornelles/3bc-lang/tree/develop-wip-wip) in which the next update will be a refactoring of 100% of the code.

if you are interested in making improvements it will be a great update!

🤠